File: C#Integration/PokerProtocol/Commands/Game/PlayerMoneyChangedCommand.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using EricUtility;
using PokerProtocol.Commands.Lobby.Response;
using PokerWorld.Game;

namespace PokerProtocol.Commands.Game
{
    public class PlayerMoneyChangedCommand : AbstractCommand
    {
        protected override string CommandName
        {
            get { return PlayerMoneyChangedCommand.COMMAND_NAME; }
        }
        public static string COMMAND_NAME = "gamePLAYER_MONEY_CHANGED";

        private readonly int m_PlayerPos;
        private readonly int m_PlayerMoney;

        public int PlayerPos
        {
            get { return m_PlayerPos; }
        }
        public int PlayerMoney
        {
            get { return m_PlayerMoney; }
        }

        public PlayerMoneyChangedCommand(StringTokenizer argsToken)
        {
            m_PlayerPos = int.Parse(argsToken.NextToken());
            m_PlayerMoney = int.Parse(argsToken.NextToken());
        }

        public PlayerMoneyChangedCommand(int pos, int money)
        {
            m_PlayerPos = pos;
            m_PlayerMoney = money;
        }

        public override void Encode(StringBuilder sb)
        {
            Append(sb, m_PlayerPos);
            Append(sb, m_PlayerMoney);
        }
    }
}

File: trunk/JAVA/PokerProtocol/src/bluffinmuffin/protocol/commands/game/PlayerMoneyChangedCommand.java
package bluffinmuffin.protocol.commands.game;

import java.util.StringTokenizer;

import bluffinmuffin.protocol.commands.AbstractCommand;

public class PlayerMoneyChangedCommand extends AbstractCommand
{
    @Override
    protected String getCommandName()
    {
        return PlayerMoneyChangedCommand.COMMAND_NAME;
    }
    
    private final int m_playerPos;
    private final int m_playerMoney;
    
    public static String COMMAND_NAME = "gamePLAYER_MONEY_CHANGED";
    
    public PlayerMoneyChangedCommand(StringTokenizer argsToken)
    {
        m_playerPos = Integer.parseInt(argsToken.nextToken());
        m_playerMoney = Integer.parseInt(argsToken.nextToken());
    }
    
    public PlayerMoneyChangedCommand(int pos, int money)
    {
        m_playerPos = pos;
        m_playerMoney = money;
    }
    
    @Override
    public void encode(StringBuilder sb)
    {
        append(sb, m_playerPos);
        append(sb, m_playerMoney);
    }
    
    public int getPlayerPos()
    {
        return m_playerPos;
    }
    
    public int getPlayerMoney()
    {
        return m_playerMoney;
    }
}
